File: API/API/Controllers/Response/VillaHotelApiResponse.cs
namespace API.Controllers.Response
{
    public class VillaHotelApiResponse<T>
    {
        public T Data { get; }

        public VillaHotelApiStatus Status { get; }

        public VillaHotelApiResponse(T data)
        {
            Data = data;
            Status = new VillaHotelApiStatus();
        }

        public VillaHotelApiResponse(int errorCode, string errorMessage)
        {
            Status = new VillaHotelApiStatus(errorCode, errorMessage);
        }
    }
}

File: HotelVillaApplication/app/src/main/java/be/cienfuegos/sandra/hotel_villa_application/api/model/VillaHotelApiResponse.java
package be.cienfuegos.sandra.hotel_villa_application.api.model;

import lombok.Getter;
import lombok.ToString;

@Getter
@ToString
public class VillaHotelApiResponse<T> {
    private T data;
    private VillaHotelApiStatus status;

    public boolean isSuccess() {
        return status.getErrorCode() == 0;
    }
}